File: src/my_project/interviews/top_150_questions_round_22/ex_82_k_smallest_element_in_bst.py
from typing import List, Union, Collection, Mapping, Optional
from abc import ABC, abstractmethod
from collections import deque

class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    def kthSmallest(self, root: TreeNode, k: int) -> int:
        # Collect all values in sorted order using in-order traversal
        values = []
        
        def inorder(node):
            if not node:
                return
            inorder(node.left)
            values.append(node.val)
            inorder(node.right)
        
        inorder(root)
        return values[k - 1]

File: src/my_project/interviews_typescript/top_150_questions_round_1/ex_82_k_smallest_element_in_bst.ts
import { TreeNode } from './TreeNode';             

function kthSmallest(root: TreeNode | null, k: number): number {
    const result: number[] = [];
    
    function inorder(node: TreeNode | null): void {
        if (!node) return;
        
        inorder(node.left);
        result.push(node.val);
        inorder(node.right);
    }
    
    inorder(root);
    return result[k - 1];
}